### PR TITLE
47372_Supervisor_view_subjects

### PR DESCRIPTION
**Pull request by Omid07**
screenshot from 2015-12-04 11 37 08

### CODE REVIEW

File: resources/views/general/subjects/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                        <strong>{{ $subject->name }}</strong>
+                        @foreach($subject->tasks as $task)
+                        <br>Task Name: {{ $task->name }}
+                        <br>Task Description: {{ $task->description }}
+                        @endforeach
+                    </td>
+                    @if(\Auth::user()->role)
+                    <td>
+                        {!! Form::open(['route' =>['subjects.edit', $subject->id], 'method' => 'GET']) !!}
+                            {!! Form::submit('Edit') !!}
```
Comment (yasin-buet) [inline resources/views/general/subjects/index.blade.php]: space after =>

File: app/Http/Controllers/SubjectsController.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $subjects = Subject::with('tasks')->orderBy('id');
+        $subjects=$subjects->paginate(1);
```
Comment (asifbuetcse) [inline app/Http/Controllers/SubjectsController.php]: space before and after =

File: resources/views/general/subjects/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@extends('layouts.supervisor')
+@section('content')
+    <div>
+        <table class="table table-striped">
+            <thead>
+                <tr>
+                    <th>ID</th>
+                    <th>Subject Name</th>
+                    @if(\Auth::user()->role)
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: space after if

File: resources/views/general/subjects/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                       <th>Actions</th>
+                    @endif 
+                </tr>
+            </thead>
+            <tbody>
+                @foreach($subjects as $subject)
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: space

File: resources/views/general/subjects/index.blade.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                <tr>
+                    <td>{{ $subject->id }}
+                    </td>
+                    <td>
+                        <strong>{{ $subject->name }}</strong>
+                        @foreach($subject->tasks as $task)
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: space

File: resources/views/general/subjects/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                        <br>Task Name: {{ $task->name }}
+                        <br>Task Description: {{ $task->description }}
+                        @endforeach
+                    </td>
+                    @if(\Auth::user()->role)
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: if

**Comment by yasin-buet** [comment]
OK

File: resources/views/general/subjects/index.blade.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                    <th>ID</th>
+                    <th>Subject Name</th>
+                    @if (\Auth::user()->role)
+                       <th>Actions</th>
+                    @endif 
+                </tr>
+            </thead>
+            <tbody>
+                @foreach ($subjects as $subject)
+                <tr>
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: 4 space for the block

File: resources/views/general/subjects/index.blade.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+                    <td>{{ $subject->id }}
+                    </td>
+                    <td>
+                        <strong>{{ $subject->name }}</strong>
+                        @foreach ($subject->tasks as $task)
+                        <br>Task Name: {{ $task->name }}
```
Comment (asifbuetcse) [inline resources/views/general/subjects/index.blade.php]: 4 space for the block

File: resources/views/general/subjects/index.blade.php [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+                <th>ID</th>
+                <th>Subject Name</th>
+                @if (\Auth::user()->role)
+                   <th>Actions</th>
+                @endif 
+            </tr>
+        </thead>
+        <tbody>
+            @foreach ($subjects as $subject)
+            <tr>
```
Comment (yasin-buet) [inline resources/views/general/subjects/index.blade.php]: 4 space

File: resources/views/general/subjects/index.blade.php [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+                </td>
+                <td>
+                    <strong>{{ $subject->name }}</strong>
+                    @foreach ($subject->tasks as $task)
+                        <br>Task Name: {{ $task->name }}
+                        <br>Task Description: {{ $task->description }}
+                    @endforeach
+                </td>
+                @if (\Auth::user()->role)
+                <td>
```
Comment (yasin-buet) [inline resources/views/general/subjects/index.blade.php]: 4 space

File: app/Http/Controllers/SubjectsController.php [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+class SubjectsController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $subjects = Subject::with('tasks')->orderBy('id')->paginate(1);
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/SubjectsController.php]: Check this comment and fix PR
https://github.com/framgia/tms_42/pull/10#discussion_r46540295
Btw, 1 is too small.

File: app/Http/Controllers/SubjectsController.php [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+        return view('general.subjects.index', compact('subjects'));
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/SubjectsController.php]: `subjects.index` is enough

File: resources/views/general/subjects/index.blade.php [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+            </tr>
+        </thead>
+        <tbody>
+            @foreach ($subjects as $subject)
+                <tr>
+                    <td>{{ $subject->id }}
+                    </td>
+                    <td>
+                        <strong>{{ $subject->name }}</strong>
+                        @foreach ($subject->tasks as $task)
```
Comment (dinhhoanglong91) [inline resources/views/general/subjects/index.blade.php]: Create subject detail page and user can click to access this detail page to see all tasks.
Check this comment
https://github.com/framgia/tms_42/pull/10#discussion_r46541585

File: resources/views/general/subjects/index.blade.php [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+                            <br>Task Name: {{ $task->name }}
+                            <br>Task Description: {{ $task->description }}
+                        @endforeach
+                    </td>
+                    @if (\Auth::user()->role)
+                        <td>
+                            {!! Form::open(['route' => ['subjects.edit', $subject->id], 'method' => 'GET']) !!}
```
Comment (dinhhoanglong91) [inline resources/views/general/subjects/index.blade.php]: https://github.com/framgia/tms_42/pull/10/files#r46541874

File: resources/views/general/subjects/index.blade.php [hunk 15]
```diff
@@ -0,0 +1,45 @@
+@extends('layouts.supervisor')
```
Comment (dinhhoanglong91) [inline resources/views/general/subjects/index.blade.php]: Same as this
https://github.com/framgia/tms_42/pull/10#discussion_r46543868

File: resources/views/general/subjects/index.blade.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+            @foreach ($subjects as $subject)
+                <tr>
+                    <td>{{ $subject->id }}
+                    </td>
+                    <td>
+                        <strong>{{ $subject->name }}</strong>
+                    </td>
+                    @if (\Auth::user()->role)
+                        <td>
+                            {!! Form::open(['route' => ['supervisor.subjects.edit', $subject->id], 'method' => 'GET']) !!}
```
Comment (dinhhoanglong91) [inline resources/views/general/subjects/index.blade.php]: Use `link_to` to link to `Edit` page.
Never use `FORM` with simple `GET` method.

File: app/Http/Controllers/Supervisor/SubjectsController.php [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $subjects = Subject::orderBy('id')->paginate(\Config::get('paginate.paginate_no'));
+        return view('general.subjects.index', compact('subjects'));
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsController.php]: `supervisor.subjects.index`

File: resources/views/supervisor/subjects/index.blade.php [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+        </thead>
+        <tbody>
+            @foreach ($subjects as $subject)
+                <tr>
+                    <td>{{ $subject->id }}
+                    </td>
+                    <td>
+                        <strong>{{ $subject->name }}</strong>
+                    </td>
+                    @if (\Auth::user()->role)
```
Comment (dinhhoanglong91) [inline resources/views/supervisor/subjects/index.blade.php]: Middleware did this for you, you don't have to check `Auth`

File: resources/views/supervisor/subjects/index.blade.php [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+    <div class="col-xs-10">
+    </div>
+</div>
+<div>
+    <table class="table table-striped">
+        <thead>
+            <tr>
+                <th>ID</th>
+                <th>Subject Name</th>
+                @if (\Auth::user()->role)
```
Comment (dinhhoanglong91) [inline resources/views/supervisor/subjects/index.blade.php]: Remove it

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**